agent/fw_wordpress.c

```diff
@@ -602,10 +608,23 @@ NR_PHP_WRAPPER(nr_wordpress_add_filter) {
   }
 
   if (true == wrap_hook) {
+    nruserfn_t* callback_wraprec;
     zval* callback = nr_php_arg_get(2, NR_EXECUTE_ORIG_ARGS);
     /* the callback here can be any PHP callable. nr_php_wrap_generic_callable
    * checks that a valid callable is passed */
-    nr_php_wrap_generic_callable(callback, nr_wordpress_wrap_hook);
+    callback_wraprec = nr_php_wrap_generic_callable(callback, nr_wordpress_wrap_hook);
+
+    // We can cheat here: wraprecs on callables are always transient, so if
+    // there's a wordpress_plugin_theme set we know it's from this transaction,
+    // and we don't have any issues around a possible multi-tenant setup.
+    if (callback_wraprec && NULL == callback_wraprec->wordpress_plugin_theme) {
```

Comment:
It isn't clear to me how this is an optimization - is `callback_wraprec->wordpress_plugin_theme` going to be set for the rest of a request after the first `add_filter` is wrapped?

Reply:
Yes, `callback_wraprec->wordpress_plugin_theme` is going to be set for the rest of a request after the first `add_filter` is wrapped.